### PR TITLE
Running Time Entry UI card

### LINKED ISSUE #1120: Timer page UI: running time entry
- [x] Show running time entry card at the bottom of the screen.
- [x] The whole card is tappable and opens edit view.
- [x] Show ticking timer.
- [x] Show description, project, task, client (if available).
- [x] Show 'Add description' if there are no description or project available.

### Running entry
timer - running

Zeplin: https://zpl.io/2vnYPq

### No description or project
timer with out timeline - running

Zeplin: https://zpl.io/adRKW61

### PR DESCRIPTION
**Pull request by kornelijepetak**
The (too) long awaited PR.

Closes #1120

### CODE REVIEW

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 1]
```diff
@@ -25,6 +27,55 @@ protected override void OnCreate(Bundle bundle)
             SetSupportActionBar(FindViewById<Toolbar>(Resource.Id.Toolbar));
             SupportActionBar.SetDisplayHomeAsUpEnabled(false);
             SupportActionBar.SetDisplayShowHomeEnabled(false);
+
+            runningEntryCardFrame = FindViewById(Resource.Id.MainRunningTimeEntryFrame);
+            playButton = FindViewById(Resource.Id.MainPlayButton);
+        }
+
+        private View runningEntryCardFrame;
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.cs]: Fields go at the beginning, please

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 2]
```diff
@@ -25,6 +27,55 @@ protected override void OnCreate(Bundle bundle)
...
+        private View playButton;
+
+        internal void OnTimeEntryCardVisibilityChanged(bool visible)
+        {
+
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.cs]: Extra line

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 3]
```diff
@@ -25,6 +27,55 @@ protected override void OnCreate(Bundle bundle)
...
+                runningEntryCardFrame.GetLocationInWindow(position);
+                var cx = runningEntryCardFrame.MeasuredWidth;
+                var cy = runningEntryCardFrame.MeasuredHeight;
+                var initialRadius = 0;
+                var finalRadius = cx;
+
+                if (!visible)
+                    (initialRadius, finalRadius) = (finalRadius, initialRadius);
+                
+                Animator animator = ViewAnimationUtils.CreateCircularReveal(runningEntryCardFrame, cx, cy, initialRadius, finalRadius);
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.cs]: `var` :P

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 4]
```diff
@@ -25,6 +27,55 @@ protected override void OnCreate(Bundle bundle)
...
+
+                if (!visible)
+                {
+                    EventHandler onAnimationEnd = null;
+                    onAnimationEnd = (o, s) => 
+                    { 
+                        runningEntryCardFrame.Visibility = ViewStates.Invisible;
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.cs]: Where are you making this guy visible again?

**Comment by kornelijepetak** [comment]
All tests are now passing (locally, and for bitrise, we'll see in a few minutes).
We have split the `Initialize` and `ViewAppeared` for some reason I can't recall which was causing the tests to fail.

I've reverted to the previous version of `Initialize` and `ViewAppeared` and I had no issues so I guess that was a problem with something else which was different before, but has since been changed.

It would be best that the functional test be repeated. Just in case.

**Pull request merged**